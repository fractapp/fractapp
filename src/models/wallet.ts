import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export function getLogo(currency: Currency) {
    let logo: ImageSourcePropType
    switch (currency) {
        case Currency.Polkadot:
            logo = require("assets/img/dot.png")
            break
        case Currency.Kusama:
            logo = require("assets/img/kusama.png")
            break
        default:
            throw ("invalid currency")
    }
    return logo
}
export function getSymbol(currency: Currency) {
    let symbol = ""
    switch (currency) {
        case Currency.Polkadot:
            symbol = "DOT"
            break
        case Currency.Kusama:
            symbol = "KSM"
            break
        default:
            throw ("invalid currency")
    }
    return symbol
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;
    usdValue: number;

    constructor(name: string, address: string, currency: Currency, balance: number, price: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;

        this.usdValue = +(balance * price).toFixed(2)
    }
}
