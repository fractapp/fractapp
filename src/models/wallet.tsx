import { ImageSourcePropType } from "react-native";

export enum Currency {
    Polkadot = 0,
    Kusama
}

export class Wallet {
    name: string;
    address: string;
    balance: number;
    currency: Currency;

    symbol: string;
    usdValue: number;
    logo: ImageSourcePropType;

    constructor(name: string, address: string, currency: Currency, balance: number) {
        this.name = name;
        this.address = address;
        this.balance = balance;
        this.currency = currency;
        
        //TODO get state
        let rate = 10
        this.usdValue = balance * rate
        switch (currency) {
            case Currency.Polkadot:
                this.logo = require("assets/img/dot.png")
                this.symbol = "DOT"
                break
            case Currency.Kusama:
                this.logo = require("assets/img/kusama.png")
                this.symbol = "KSM"
                break
        }

    }
}
