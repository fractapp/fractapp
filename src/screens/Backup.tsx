import React, {useContext, useEffect} from 'react';
import {
  StyleSheet,
  View,
  ScrollView,
  Text,
  FlatList,
  Image,
  TouchableOpacity,
} from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Dialog from 'storage/Dialog';
import BackupUtils from 'utils/backup';
import db from 'utils/db';
import backupUtil from 'utils/backup';

export const Backup = ({navigation, route}: {navigation: any; route: any}) => {
  const dialogContext = useContext(Dialog.Context);

  const onSuccess = () => {
    navigation.reset({
      index: 0,
      routes: [{name: 'Home'}],
    });
    dialogContext.dispatch(Dialog.close());
  };

  useEffect(() => {
    if (route.params?.isSuccess == null) {
      return;
    }

    const dir =
      route.params.type == BackupUtils.BackupType.File
        ? 'Downloads'
        : backupUtil.GoogleDriveFolder;

    dialogContext.dispatch(
      Dialog.open(
        'Success save wallet',
        `If you lose access to file then you will not be able to restore access to the wallet. File "${route.params?.fileName}.json" saved in "${dir}" directory`,
        onSuccess,
      ),
    );
  }, [route.params?.isSuccess]);

  const backupFile = async () => {
    const seed = await db.getSeed();
    await BackupUtils.backupFile(
      () =>
        navigation.navigate('WalletFileBackup', {
          seed: seed?.split(' '),
          type: BackupUtils.BackupType.File,
          callerScreen: 'Backup',
        }),
      () =>
        dialogContext.dispatch(
          Dialog.open(
            'Open settings',
            'If you want to save a file then open the application settings and give it access to the storage.',
            () => dialogContext.dispatch(Dialog.close()),
          ),
        ),
    );
  };
  const backupGoogleDrive = async () => {
    const seed = await db.getSeed();
    await BackupUtils.backupGoogleDrive(() =>
      navigation.navigate('WalletFileBackup', {
        seed: seed?.split(' '),
        type: BackupUtils.BackupType.GoogleDrive,
        callerScreen: 'Backup',
      }),
    );
  };
  const backupSeed = async () => {
    const seed = await db.getSeed();
    navigation.navigate('SaveSeed', {
      seed: seed?.split(' '),
      onSuccess: () =>
        navigation.reset({
          index: 0,
          routes: [{name: 'Home'}],
        }),
    });
  };

  const menuItems = [
    {
      img: (
        <Image
          source={require('assets/img/google-drive.png')}
          style={{width: 32, height: 32}}
        />
      ),
      title: 'Google disk',
      onClick: () => backupGoogleDrive(),
    },
    {
      img: <MaterialIcons name="folder-open" size={32} color="#888888" />,
      title: 'File to phone',
      onClick: () => backupFile(),
    },
    {
      img: (
        <MaterialCommunityIcons name="content-copy" size={32} color="#888888" />
      ),
      title: 'Backup seed',
      onClick: () => backupSeed(),
    },
  ];
  const renderItem = ({item}: {item: any}) => (
    <TouchableOpacity style={styles.menuItem} onPress={item.onClick}>
      {item.img}
      <Text style={styles.menuTitle}>{item.title}</Text>
    </TouchableOpacity>
  );

  return (
    <View style={styles.profile}>
      <ScrollView showsVerticalScrollIndicator={false} style={{width: '100%'}}>
        <View style={{alignItems: 'center', justifyContent: 'center'}}>
          <Text style={styles.description}>
            Save your wallet in a safe place. If you lose your wallet, you
            cannot restore access to it.
          </Text>
          <FlatList
            style={styles.menu}
            ItemSeparatorComponent={() => <View style={styles.dividingLine} />}
            data={menuItems}
            renderItem={renderItem}
            keyExtractor={(item) => item.title}
          />
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  profile: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  dividingLine: {
    alignSelf: 'center',
    width: '100%',
    borderBottomWidth: 1,
    borderBottomColor: '#DADADA',
  },
  description: {
    width: '90%',
    marginTop: 10,
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    color: '#888888',
  },
  menu: {
    flex: 1,
    width: '88%',
    marginTop: 20,
  },
  menuItem: {
    flexDirection: 'row',
    paddingTop: 10,
    paddingBottom: 10,
  },
  menuTitle: {
    marginLeft: 10,
    fontSize: 17,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    color: 'black',
    alignSelf: 'center',
  },
});
