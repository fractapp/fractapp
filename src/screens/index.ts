export {ImportWallet} from './ImportWallet';
export {SaveSeed} from './SaveSeed';
export {SaveWallet} from './SaveWallet';
export {SettingWallet} from './SettingWallet';
export {Start} from './Start';
export {WalletDetails} from './WalletDetails';
export {WalletFileBackup} from './WalletFileBackup';
export {Wallets} from './Wallets';
export {Settings} from './Settings';
export {ConfirmSaveSeed} from './ConfirmSaveSeed';
export {ImportSeed} from './ImportSeed';
export {WalletFileImport} from './WalletFileImport';
export {GoogleDrivePicker} from './GoogleDrivePicker';
export {Receive} from './Receive';
export {Milestone} from './Milestone';
export {TransactionDetails} from './TransactionDetails';
export {SplashScreen} from './SplashScreen';
export {Backup} from './Backup';
export {NewPassCode} from './NewPassCode';
export {VerifyPassCode} from './VerifyPassCode';
export {EditProfile} from './EditProfile';
export {EditPhoneNumber} from './EditPhoneNumber';
export {SelectCountry} from './SelectCountry';
export {ConfirmCode} from './ConfirmCode';
export {EditEmail} from './EditEmail';
export {Navigation} from './Navigation';
export {Chats} from './Chats';
export {Chat} from './Chat';
export {Connecting} from './Connecting';
export {EditUsername} from './EditUsername';
export {EditName} from './EditName';
export {Search} from './Search';
export {SelectWallet} from './SelectWallet';
export {Send} from './Send';
export {EnterAmount} from './EnterAmount';
export {SendWithEnterAddress} from './SendWithEnterAddress';
