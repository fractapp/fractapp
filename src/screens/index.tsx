export { default as ImportWallet } from './ImportWallet';
export { default as SaveSeed } from './SaveSeed';

export { default as SaveWallet } from './SaveWallet';
export { default as SettingWallet } from './SettingWallet';
export { default as Start } from './Start';

export { default as WalletDetails } from './WalletDetails';

export { default as WalletEncryption } from './WalletEncryption';
export { default as Wallets } from './Wallets';
export { ConfirmSaveSeed } from './ConfirmSaveSeed'