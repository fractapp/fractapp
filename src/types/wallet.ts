import { ImageSourcePropType } from 'react-native';
import { AccountType, Network } from 'types/account';
import { Transaction, TxAction } from 'types/transaction';

/**
 * @category Models
 */
export enum Currency {
  DOT = 0,
  KSM,
}


/**
 * get network by currency
 * @category Models
 */

export function getNetwork(currency: Currency): Network {
  switch (currency) {
    case Currency.DOT:
      return Network.Polkadot;
    case Currency.KSM:
      return Network.Kusama;
    default:
      return Network.Polkadot;
  }
}


/**
 * get logo for wallet
 * @category Models
 */

export function getFullCurrencyName(currency: Currency) {
  let name = '';
  switch (currency) {
    case Currency.DOT:
      name = 'Polkadot';
      break;
    case Currency.KSM:
      name = 'Kusama';
      break;
    default:
      throw new Error('invalid currency');
  }
  return name;
}

export function getColor(currency: Currency) {
  let color = '';
  switch (currency) {
    case Currency.DOT:
      color = '#E6007A';
      break;
    case Currency.KSM:
      color = '#434C74';
      break;
    default:
      throw new Error('invalid currency');
  }
  return color;
}

export function getLogo(currency: Currency) {
  let logo: ImageSourcePropType;
  switch (currency) {
    case Currency.DOT:
      logo = require('assets/img/dot.png');
      break;
    case Currency.KSM:
      logo = require('assets/img/kusama.png');
      break;
    default:
      throw new Error('invalid currency');
  }
  return logo;
}

/**
 * check logo without border
 * @category Models
 */
export function withoutBorder(currency: Currency): boolean {
  return currency === Currency.DOT || currency === Currency.KSM;
}

/**
 * get symbol for wallet
 * @category Models
 */
export function getSymbol(currency: Currency) {
  let symbol = '';
  switch (currency) {
    case Currency.DOT:
      symbol = 'DOT';
      break;
    case Currency.KSM:
      symbol = 'KSM';
      break;
    default:
      throw new Error('invalid currency');
  }
  return symbol;
}

/**
 * string to currency
 * @category Models
 */
export function toCurrency(currency: string): Currency {
  switch (currency) {
    case 'DOT':
      return Currency.DOT;
    case 'KSM':
      return Currency.KSM;
    default:
      throw new Error('invalid currency');
  }
}


/**
 * currency to string
 * @category Models
 */
export function fromCurrency(currency: Currency): string {
  switch (currency) {
    case Currency.DOT:
      return 'DOT';
    case Currency.KSM:
      return 'KSM';
    default:
      throw new Error('invalid currency');
  }
}

/**
 * filter Tx by wallet
 * @category Models
 */
export function filterTxsByAccountType(txs: Array<Transaction>, action: AccountType): Array<Transaction>  {
  switch (action) {
    case AccountType.Main:
      return txs.filter((tx) => tx.action === TxAction.Transfer || tx.action === TxAction.Staking || tx.action === TxAction.StakingWithdrawn);
    case AccountType.Staking:
      return txs.filter((tx) => tx.action === TxAction.Staking || tx.action === TxAction.StakingWithdrawn || tx.action === TxAction.StakingReward);
    default:
      throw new Error('invalid action type');
  }
}
