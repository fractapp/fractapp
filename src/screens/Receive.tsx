import React from 'react';
import {StyleSheet, View, Image, Text, TouchableHighlight} from 'react-native';
import Clipboard from '@react-native-community/clipboard';
import {Currency, getSymbol} from 'types/wallet';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import QRCode from 'react-native-qrcode-svg';
import Share from 'react-native-share';
import {showMessage} from 'react-native-flash-message';

/**
 * Screen with receiver information and qr code
 * @category Screens
 */
export const Receive = ({route}: {route: any}) => {
  const address: string = route.params.address;
  const currency: Currency = route.params.currency;

  return (
    <View style={{flex: 1, alignItems: 'center'}}>
      <View style={styles.qrCode}>
        <View style={styles.title}>
          <Image
            source={require('assets/img/logo.png')}
            style={{width: 30, height: 30, marginRight: 10}}
          />
          <Text style={styles.titleText}>Fractapp Wallet</Text>
        </View>
        <QRCode value={address} size={250} />
        <Text style={styles.address}>{address}</Text>
      </View>
      <View style={styles.btns}>
        <View>
          <TouchableHighlight
            testID="copyBtn"
            onPress={() => {
              Clipboard.setString(address);
              showMessage({
                message: 'Address copied',
                type: 'info',
                icon: 'info',
              });
            }}
            underlayColor="#76C7E3"
            style={styles.btn}>
            <MaterialCommunityIcons
              name="content-copy"
              size={24}
              color="white"
            />
          </TouchableHighlight>
          <Text style={styles.btnText}>Copy</Text>
        </View>
        <View style={{marginLeft: 30}}>
          <TouchableHighlight
            testID="shareBtn"
            onPress={() =>
              Share.open({
                url: `My address for ${getSymbol(currency)}: ${address}`,
                type: 'text/plain',
              })
            }
            underlayColor="#76C7E3"
            style={[styles.btn]}>
            <MaterialCommunityIcons
              name="upload-outline"
              size={24}
              color="white"
            />
          </TouchableHighlight>
          <Text style={styles.btnText}>Share</Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  btns: {
    marginTop: 30,
    justifyContent: 'center',
    flexDirection: 'row',
  },
  btn: {
    width: 60,
    height: 60,
    backgroundColor: '#2AB2E2',
    borderRadius: 45,
    alignItems: 'center',
    justifyContent: 'center',
  },
  btnText: {
    paddingTop: 2,
    fontSize: 16,
    fontFamily: 'Roboto-Regular',
    fontStyle: 'normal',
    textAlign: 'center',
    color: 'black',
  },
  qrCode: {
    marginTop: 100,
    width: '80%',
    padding: 10,
    borderWidth: 0.5,
    borderColor: '#BFBDBD',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
  },
  title: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  titleText: {
    marginTop: 6,
    marginBottom: 10,
    textAlign: 'center',
    fontSize: 20,
    fontFamily: 'Roboto-Medium',
    color: '#2AB2E2',
  },
  address: {
    marginTop: 20,
    textAlign: 'center',
    fontSize: 15,
    fontFamily: 'Roboto-Regular',
    color: '#888888',
  },
});
