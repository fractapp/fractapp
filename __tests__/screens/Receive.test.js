import React, { useContext } from 'react';
import {Receive} from 'screens/Receive';
import renderer from 'react-test-renderer';
import {Currency, getSymbol} from 'types/wallet';
import {render, fireEvent} from '@testing-library/react-native';
import Clipboard from '@react-native-community/clipboard';
import {showMessage} from 'react-native-flash-message';
import Share from 'react-native-share';
import GlobalStore from 'storage/Global';

jest.mock('storage/DB', () => ({}));

jest.mock('react-native-qrcode-svg', () => ({value, size}) => {
  const React = require('react');
  const ReactNative = require('react-native');
  return (
    <ReactNative.View>
      <ReactNative.Text>{value}</ReactNative.Text>
      <ReactNative.Text>{size}</ReactNative.Text>
    </ReactNative.View>
  );
});
jest.mock('react-native-share', () => ({
  open: jest.fn(),
}));
jest.mock('@react-native-community/clipboard', () => ({
  setString: jest.fn(),
}));
jest.mock('react-native-flash-message', () => ({
  showMessage: jest.fn(),
}));

jest.mock('react-native-i18n', () => ({
  t: (value) => value,
}));

it('Test receive #1', () => {
  const tree = renderer
    .create(
      <Receive
        route={{
          params: {
            address: '1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE',
            currency: Currency.DOT,
          },
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test receive #2', () => {
  const params = {
    address: 'F7Zbj7rRJQLQVvQn8tKnWmSByAoFFYKYd17hSxdmYbcZzoE',
    currency: Currency.KSM,
  };

  const tree = renderer.create(<Receive route={{params: params}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test receive click copy', () => {
  const params = {
    address: 'F7Zbj7rRJQLQVvQn8tKnWmSByAoFFYKYd17hSxdmYbcZzoE',
    currency: Currency.KSM,
  };

  const component = render(<Receive route={{params: params}} />);
  fireEvent.press(component.getByTestId('copyBtn'));
  expect(Clipboard.setString).toBeCalledWith(params.address);
  expect(showMessage).toBeCalledWith({
    message: 'show_msg.address_copied',
    type: 'info',
    icon: 'info',
  });
});

it('Test receive click share', () => {
  const params = {
    address: 'F7Zbj7rRJQLQVvQn8tKnWmSByAoFFYKYd17hSxdmYbcZzoE',
    currency: Currency.KSM,
    shareLink: 'https://send.fractapp.com/send.html?user=F7Zbj7rRJQLQVvQn8tKnWmSByAoFFYKYd17hSxdmYbcZzoE&type=address&currency=KSM',
  };

  const component = render(<Receive route={{params: params}} />);
  fireEvent.press(component.getByTestId('shareBtn'));
  expect(Share.open).toBeCalledWith({
    url: `share_text ${getSymbol(params.currency)}: ${params.address}\n\nshare_link: ${params.shareLink}`,
    type: 'text/plain',
    title: '',
    message: '',
  });
});
