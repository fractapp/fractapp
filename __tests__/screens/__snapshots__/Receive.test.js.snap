// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test receive #1 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#BFBDBD",
        "borderRadius": 10,
        "borderWidth": 0.5,
        "justifyContent": "center",
        "marginTop": 100,
        "padding": 10,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/logo.png",
          }
        }
        style={
          Object {
            "height": 30,
            "marginRight": 10,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Medium",
            "fontSize": 20,
            "marginBottom": 10,
            "marginTop": 6,
            "textAlign": "center",
          }
        }
      >
        fractapp_wallet_title
      </Text>
    </View>
    <View>
      <Text>
        1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE
      </Text>
      <Text>
        250
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#888888",
          "fontFamily": "Roboto-Regular",
          "fontSize": 15,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      1exaAg2VJRQbyUBAeXcktChCAqjVP9TUxF3zo23R2T6EGdE
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2AB2E2",
            "borderRadius": 45,
            "height": 60,
            "justifyContent": "center",
            "width": 60,
          }
        }
        testID="copyBtn"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          󰆏
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 16,
            "fontStyle": "normal",
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        copy_btn
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "marginLeft": 30,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#2AB2E2",
              "borderRadius": 45,
              "height": 60,
              "justifyContent": "center",
              "width": 60,
            },
          ]
        }
        testID="shareBtn"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          󰸇
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 16,
            "fontStyle": "normal",
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        share_btn
      </Text>
    </View>
  </View>
</View>
`;

exports[`Test receive #2 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#BFBDBD",
        "borderRadius": 10,
        "borderWidth": 0.5,
        "justifyContent": "center",
        "marginTop": 100,
        "padding": 10,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/img/logo.png",
          }
        }
        style={
          Object {
            "height": 30,
            "marginRight": 10,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#2AB2E2",
            "fontFamily": "Roboto-Medium",
            "fontSize": 20,
            "marginBottom": 10,
            "marginTop": 6,
            "textAlign": "center",
          }
        }
      >
        fractapp_wallet_title
      </Text>
    </View>
    <View>
      <Text>
        F7Zbj7rRJQLQVvQn8tKnWmSByAoFFYKYd17hSxdmYbcZzoE
      </Text>
      <Text>
        250
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#888888",
          "fontFamily": "Roboto-Regular",
          "fontSize": 15,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      F7Zbj7rRJQLQVvQn8tKnWmSByAoFFYKYd17hSxdmYbcZzoE
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2AB2E2",
            "borderRadius": 45,
            "height": 60,
            "justifyContent": "center",
            "width": 60,
          }
        }
        testID="copyBtn"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          󰆏
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 16,
            "fontStyle": "normal",
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        copy_btn
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "marginLeft": 30,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#2AB2E2",
              "borderRadius": 45,
              "height": 60,
              "justifyContent": "center",
              "width": 60,
            },
          ]
        }
        testID="shareBtn"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          󰸇
        </Text>
      </View>
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Roboto-Regular",
            "fontSize": 16,
            "fontStyle": "normal",
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        share_btn
      </Text>
    </View>
  </View>
</View>
`;
